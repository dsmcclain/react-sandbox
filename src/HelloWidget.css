.HelloWidget {
    border: 2px solid #555;
    text-align: center;
    background: #FFFE90;
    color: #333;
    margin: 20px;
    padding: 20px;
  }