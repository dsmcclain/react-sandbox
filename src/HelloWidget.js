import React, { Component } from 'react';
import './HelloWidget.css';

class HelloWidget extends Component {
  constructor(props) {
      super(props);
      this.state = { 
        greeting: 'Hello',
        AbbreviationCount: 0,
      };
      this.abbreviate = this.abbreviate.bind(this);
      this.removePeople = this.removePeople.bind(this);
  }

  abbreviate() {
      const fredness = this.state.greeting;
      if (fredness === 'Hello') {
        this.setState({ greeting: 'Hi' });
        this.setState({ AbbreviationCount: this.state.AbbreviationCount + 1 });
      } else {
        this.setState({ greeting: 'Hello' });
      };
  }

  removePeople() {
      this.props.removePeople(this.props.name);
  }
  
  render() {
    return (
      <div className="HelloWidget">
        {this.state.greeting} {this.props.name}!
        <br />
        This greeting has been abbreviated {this.state.AbbreviationCount} times.
        <br />
        <button className="abbreviate" onClick={this.abbreviate}>Abbreviate</button>
        <button className="remove" onClick={this.removePeople}>Remove Me!</button>
      </div>
    );
  }
}

export default HelloWidget;