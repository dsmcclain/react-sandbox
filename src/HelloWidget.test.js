import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import HelloWidget from './HelloWidget';

describe(HelloWidget, () => {
    const name = 'Person';
    const mockRemovePeople = jest.fn();
    const component = shallow(
        <HelloWidget name={name} removePeople={mockRemovePeople}/>
    );

    it ('renders and matches our snapshot', () => {
        const component = renderer.create(
            <HelloWidget name="Person" />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('contains the supplied name', () => {
        expect(component.text()).toContain(name);
    });

    it('modifies the greeting when abbreviate button is clicked', () => {
        component.find('button.abbreviate').simulate('click');
        expect(component.text()).toContain('Hi');
    });

    it('calls the passed in removePeople function when remove button is clicked', () => {
        component.find('button.remove').simulate('click');
        expect(mockRemovePeople).toBeCalled();
    });
});