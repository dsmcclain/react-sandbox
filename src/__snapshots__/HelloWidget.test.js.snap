// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWidget renders and matches our snapshot 1`] = `
<div
  className="HelloWidget"
>
  Hello
   
  Person
  !
  <br />
  This greeting has been abbreviated 
  0
   times.
  <br />
  <button
    className="abbreviate"
    onClick={[Function]}
  >
    Abbreviate
  </button>
  <button
    className="remove"
    onClick={[Function]}
  >
    Remove Me!
  </button>
</div>
`;
